CRM/Contribute/Form/ContributionBase.php

```diff
@@ -293,7 +293,7 @@ public function preProcess() {
       $isPayLater = CRM_Utils_Array::value('is_pay_later', $this->_values);
 
       if ($isMonetary &&
-        (!$isPayLater || !empty($this->_values['payment_processor']))
+        (!$isPayLater && !empty($this->_values['payment_processor']))
```

Comment:
My sense is that it should have been valid to have $isMonetary==TRUE && $isPayLater==TRUE && empty($this->_values['payment_processor']) up until now, and that going forward, $isPayLater==TRUE should ensure that $this->_value['payment_processor'] holds the new Manual process class. It seems like you now want this part of the code to make $isPayLater==TRUE to cause $this->_value['payment_processor'] to be filled. Is that what you are aiming at here?